### ISSUE TITLE
Enforce use of modern syntax using unicorn plugin

### ISSUE THREAD

**Issue by jeddy3** [stylelint/eslint-config-stylelint#77]
Use `String.includes`. Ref:  https://github.com/stylelint/stylelint/pull/4464

> Unfortunately, I'm not aware of an ESLint rule to enforce this. Only https://github.com/sindresorhus/eslint-plugin-unicorn

I think that @XhmikosR's PRs to move to modern functions like `include`, `startsWith` etc has made the code more understandable. I'm very much in favour of enforcing good practise with tooling and would be happy to see unicorn used in this config. We'd only turn on the rules that support @XhmikosR's changes, though.

Thoughts?

**Comment by XhmikosR** [comment]
Hehe, I already use xo TBH :)

https://github.com/XhmikosR/stylelint/tree/xo

Adapted so that things work with prettier enforced too

**Comment by XhmikosR** [comment]
Keep in mind this branch is old and can definitely be tweaked further. I just tried to leverage xo and backport any changes in smaller batches.

**Comment by XhmikosR** [comment]
And this was the main upstream for includes PR https://github.com/stylelint/stylelint/pull/4417

**Comment by jeddy3** [comment]
To move this discussion forward, it would be great to see how many errors there are when using the `unicorn` plugin on the main repo.

**Comment by XhmikosR** [comment]
@jeddy3 https://github.com/stylelint/stylelint/compare/xo

I can rebase it tomorrow and you can play with it by enabling more rules which currently I have them disabled so that it follows the current style.

**Comment by XhmikosR** [comment]
@jeddy3 I rebased it, but keep in mind it's using prettier 1.x hence the whitespace changes.

https://github.com/stylelint/stylelint/compare/xo

That being said,

1. we'd need to include the jest config too
2. adapt this to our needs

**Comment by XhmikosR** [comment]
I rebased the branch https://github.com/stylelint/stylelint/compare/v14...xo?w=1

I have fixed the prettier issue by inlining the prettier config. So the reported errors are more valid now. That being said there are plenty of missing rules errors and the TypeScript ones don't look all of them correct. Plus there are 2-3 false autofixes.

Anyway, if anyone has the time feel free to experiment with the branch and see what else can be tweaked.
